### PR TITLE
[docs][location][permissions]: Update expo-location doc and permissions  guide

### LINKED ISSUE #17183: [docs] [expo-location] Missing documentation about LocationAccuracy & Permissions
### Summary

In https://docs.expo.dev/versions/latest/sdk/location, there is some documentation about Permissions and Usage, but there is something important missing:

I don't know when to use COARSE, and when to use FINE location.

In my code I have:
```
Location.getCurrentPositionAsync({
  accuracy: Location.LocationAccuracy.Balanced,
}
```

I don't know if my permissions should be this:
```
      "permissions": [
        "ACCESS_COARSE_LOCATION",
        "ACCESS_FINE_LOCATION"
      ],
```

or this

```
      "permissions": [
        "ACCESS_COARSE_LOCATION"
      ],
```

If you check this: `expo-location/build/Location.types.d.ts`, you will see:
```
/**
 * Enum with available location accuracies.
 */
export declare enum LocationAccuracy {
    /**
     * Accurate to the nearest three kilometers.
     */
    Lowest = 1,
    /**
     * Accurate to the nearest kilometer.
     */
    Low = 2,
    /**
     * Accurate to within one hundred meters.
     */
    Balanced = 3,
    /**
     * Accurate to within ten meters of the desired target.
     */
    High = 4,
    /**
     * The best level of accuracy available.
     */
    Highest = 5,
    /**
     * The highest possible accuracy that uses additional sensor data to facilitate navigation apps.
     */
    BestForNavigation = 6
}
```

But... It's not clear WHEN I have to add the `_FINE_` permission. Is it necessary for `BestForNavigation`? for `Highest`? for `High`? Maybe I'm adding `_FINE_` and I don't really need it.

### Link to the related docs page

https://docs.expo.dev/versions/latest/sdk/location

### Anything else?

Love you all

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

The `expo-location` docs were missing info on how to exclude permission from the **app.json** file. It also doesn't explain, what are all the permissions available, and each permission's description.

The [excluding permission for Android docs](https://docs.expo.dev/guides/permissions/#excluding-android-permissions) are using classic build way of removing a specific permission from **app.json** file.

Also, Refs https://github.com/expo/expo/issues/17183

# How

This PR updates the following pages:
- /versions/latest/sdk/location/
  - Use `AndroidPermissions` to list down all expo-location permissions and their descriptions
  - Add a section to exclude a specific permission and redirect to /guides/permissions/#excluding-android-permissions section
  - backport SDK 46 & 45
- /guides/permissions
  - /#excluding-android-permissions 
    - ~~update the section to include info about how to exclude a permission using Config Plugin~~
    - add an example on how to use exclude the permission (uses `expo-locaiton` example to exclude `ACCESS_FINE_LOCATION`)
  - Updates **Andoird** -> Managed workflow & **Android** -> Bare workflow sections to remove classic build permission omitting info
  - Move info related to classic build and bare native apps into collapsible sections
  - Also added an example that refers to [Android documentation on Location Accuracy values](https://developer.android.com/training/location/permissions#accuracy) to when to exclude a permission based on the Enum Accuracy values 

# Test Plan

Changes have been tested by running the `docs/` locally.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by Simek — APPROVED** [review]
Thank you for the attention to details in this update and backporting the changes to older SDKs! 🙏

**Review by byCedric — APPROVED** [review]

File: docs/pages/guides/permissions.md [hunk 1]
```diff
@@ -26,30 +28,54 @@ You can find the full list of available properties in [Apple's InfoPlistKeyRefer
...
+
+When you are building your app with [EAS Build](/build/introduction/), you can use a [Config Plugin](/guides/config-plugins/#using-a-plugin-in-your-app) in **app.json** or **app.config.json** to exclude a permission. In the example config below, the `ACCESS_FINE_LOCATION` is omitted when the app is built using EAS:
+
+```json
+{
+  "expo": {
+    "plugins": [
+      "expo-location": {
+        "withBlockedPermission": ["ACCESS_FINE_LOCATION"]
+      }
```
Comment (byCedric) [inline docs/pages/guides/permissions.md:59]: Oh wait! This one is incorrect, we don't have the `withBlockedPermission` option in the `expo-location`'s config plugin: https://github.com/expo/expo/blob/main/packages/expo-location/plugin/src/withLocation.ts#L24-L30.

And it should be in this format:
```json
{
  "expo": {
    "plugins": [
      "expo-location",
      { ...options }
    ]
  }
}
```

Comment (byCedric) [reply docs/pages/guides/permissions.md:59]: I also see your point in using the blocked android permissions, but we have to do either one of this:
1. Make the `withBlockedAndroidPermissions` a separate config plugin
2. Add the option, like `withoutAndroidPermissions`, to each config plugin

**Review by byCedric — CHANGES_REQUESTED** [review]
See comment above 😄 Everything else is 🌈✨

File: docs/pages/guides/permissions.md (hunk 1 above)
Comment (lukmccall) [reply docs/pages/guides/permissions.md:59]: I think that is worth doing. However, I don't have a strong opinion on which approach will be the best. I'm okay with both solutions.

Comment (amandeepmittal) [reply docs/pages/guides/permissions.md:59]: Sorry for not getting back here 😅 

I am failing to understand to exclude permission using a Config plugin vs the [docs currently recommending](https://docs.expo.dev/distribution/app-stores/#android-permissions) which is to add `android.blockedPermissions` array. I think we need to have "best practice" approach if there are multiple ways to do it in the docs (as to not create any confusion).

**Comment by amandeepmittal** [comment]
Hey @byCedric, can you please re-review this PR? I've made the following updates from the feedback I got and to tackle the existing confusion around excluding permission and Location Accuracy enum values mentioned in `expo-location` API reference docs.
- Location (expo-location) API reference: Added an example that refers to [Android documentation on Location Accuracy values](https://developer.android.com/training/location/permissions#accuracy) to when to exclude a permission based on the Enum Accuracy values  
- Permission:  Remove config plugin example for EAS Build and direct to the user to https://docs.expo.dev/distribution/app-stores/#android-permissions

File: docs/pages/guides/permissions.md [hunk 2]
```diff
@@ -26,30 +28,40 @@ You can find the full list of available properties in [Apple's InfoPlistKeyRefer
...
 
-To limit the permissions your managed workflow app requires, set the `android.permissions` property in your [**app.json** file](../../workflow/configuration.md#android) to list only the permissions you need, and Expo will also include the minimum permissions it requires to run. See the [`Permission types`](#permission-types) below to learn about which Android permissions are added. You can find a full list of all available permissions in the [Android Manifest.permissions reference](https://developer.android.com/reference/android/Manifest.permission).
+Some Expo and React Native modules include permissions by default. For example, if you use `expo-location`, the `ACCESS_COARSE_LOCATION`, `ACCESS_FINE_LOCATION` and `FOREGROUND_SERVICE` have added to your app's permissions automatically. [See the `android.permissions` documentation](/versions/latest/config/app.md#permissions) to learn about which permissions are always included. To limit the permissions your managed workflow app requires, see [Excluding Android Permissions](#excluding-android-permissions).
```
Comment (Simek) [inline docs/pages/guides/permissions.md:33]:
```suggestion
Some Expo and React Native modules include permissions by default. For example, if you use `expo-location`, the `ACCESS_COARSE_LOCATION`, `ACCESS_FINE_LOCATION` and `FOREGROUND_SERVICE` permissions will be added to your app's automatically. [See the `android.permissions` documentation](/versions/latest/config/app.md#permissions) to learn about which permissions are always included. To limit the permissions your managed workflow app requires, see [Excluding Android Permissions](#excluding-android-permissions).
```

Wording nit, not sure if my version is fully correct but befor the change the sentence sound off to me.

Comment (amandeepmittal) [reply docs/pages/guides/permissions.md:33]: This is a good change. Thanks for catching it 👍

File: docs/pages/versions/unversioned/sdk/location.md [hunk 3]
```diff
@@ -21,20 +22,31 @@ import SnackInline from '~/components/plugins/SnackInline';
...
+The `expo-location` module requires permission for approximate (`ACCESS_COARSE_LOCATION`) and precise (`ACCESS_FINE_LOCATION`) device location in the `android.permission` property in **app.json**. It also needs foreground service permission (`FOREGROUND_SERVICE`) to subscribe to location updates while the app is in use. When you install this module in your Expo project, all of the three permissions are added automatically.
+
+To use background location features, you must add the `ACCESS_BACKGROUND_LOCATION` and [submit your app for review and request access to use the background location permission](https://support.google.com/googleplay/android-developer/answer/9799150?hl=en).
+
+<AndroidPermissions permissions={['ACCESS_COARSE_LOCATION', 'ACCESS_FINE_LOCATION', 'FOREGROUND_SERVICE', 'ACCESS_BACKGROUND_LOCATION']} />
+
+#### Excluding a permission
+
+When your Expo project doesn't benefit from having particular permission included, you can omit it. For example, if your application doesn't need access to the precise location, you can exclude the `ACCESS_FINE_LOCATION` permission.
```
Comment (byCedric) [inline docs/pages/versions/unversioned/sdk/location.md:33]: Maybe it's good to mention that excluding "required permissions" from modules can break functionality related to those permissions. That way, people are sort of warned before doing so.

File: docs/pages/versions/v45.0.0/sdk/location.md [hunk 4]
```diff
@@ -21,20 +22,31 @@ import SnackInline from '~/components/plugins/SnackInline';
...
+The `expo-location` module requires permission for approximate (`ACCESS_COARSE_LOCATION`) and precise (`ACCESS_FINE_LOCATION`) device location in the `android.permission` property in **app.json**. It also needs foreground service permission (`FOREGROUND_SERVICE`) to subscribe to location updates while the app is in use. When you install this module in your Expo project, all of the three permissions are added automatically.
+
+To use background location features, you must add the `ACCESS_BACKGROUND_LOCATION` and [submit your app for review and request access to use the background location permission](https://support.google.com/googleplay/android-developer/answer/9799150?hl=en).
+
+<AndroidPermissions permissions={['ACCESS_COARSE_LOCATION', 'ACCESS_FINE_LOCATION', 'FOREGROUND_SERVICE', 'ACCESS_BACKGROUND_LOCATION']} />
+
+#### Excluding a permission
+
+When your Expo project doesn't benefit from having particular permission included, you can omit it. For example, if your application doesn't need access to the precise location, you can exclude the `ACCESS_FINE_LOCATION` permission.
```
Comment (byCedric) [inline docs/pages/versions/v45.0.0/sdk/location.md:33]: Same here, maybe good to warn that excluding required permissions can break functionality of the API

File: docs/pages/versions/v46.0.0/sdk/location.md [hunk 5]
```diff
@@ -21,20 +22,31 @@ import SnackInline from '~/components/plugins/SnackInline';
...
+The `expo-location` module requires permission for approximate (`ACCESS_COARSE_LOCATION`) and precise (`ACCESS_FINE_LOCATION`) device location in the `android.permission` property in **app.json**. It also needs foreground service permission (`FOREGROUND_SERVICE`) to subscribe to location updates while the app is in use. When you install this module in your Expo project, all of the three permissions are added automatically.
+
+To use background location features, you must add the `ACCESS_BACKGROUND_LOCATION` and [submit your app for review and request access to use the background location permission](https://support.google.com/googleplay/android-developer/answer/9799150?hl=en).
+
+<AndroidPermissions permissions={['ACCESS_COARSE_LOCATION', 'ACCESS_FINE_LOCATION', 'FOREGROUND_SERVICE', 'ACCESS_BACKGROUND_LOCATION']} />
+
+#### Excluding a permission
+
+When your Expo project doesn't benefit from having particular permission included, you can omit it. For example, if your application doesn't need access to the precise location, you can exclude the `ACCESS_FINE_LOCATION` permission.
```
Comment (byCedric) [inline docs/pages/versions/v46.0.0/sdk/location.md:33]: Same here, maybe good to warn that excluding required permissions can break functionality of the API

**Review by byCedric — APPROVED** [review]
Left some optional comments, looking good

File: docs/pages/guides/permissions.md [hunk 6]
```diff
@@ -26,30 +28,40 @@ You can find the full list of available properties in [Apple's InfoPlistKeyRefer
...
+Apps using dangerous or signature permissions without valid reasons _may be rejected by Google_. Make sure you follow the [Android permissions best practices](https://developer.android.com/training/permissions/usage-notes) when submitting your app.
 
 ### Bare workflow
 
 In the bare workflow, permissions are controlled in your project **AndroidManifest.xml**.
 
-Some Expo and React Native modules include permissions by default. If you use `expo-location`, for example, both the `ACCESS_COARSE_LOCATION` and `ACCESS_FINE_LOCATION` are implied and added to your app's permissions automatically. To limit the permissions your managed workflow app requires, add them them to a [list of explicitly excluded permissions](#excluding-android-permissions).
+Some Expo and React Native modules include permissions by default. For example, if you use `expo-location`, the `ACCESS_COARSE_LOCATION`, `ACCESS_FINE_LOCATION` and `FOREGROUND_SERVICE` are added to your app's permissions automatically. To limit the permissions your bare workflow app requires, see [Excluding Android Permissions](#excluding-android-permissions).
```
Comment (brentvatne) [inline docs/pages/guides/permissions.md:41]:
```suggestion
Some Expo and React Native modules include permissions by default. For example, if you use `expo-location`, the `ACCESS_COARSE_LOCATION`, `ACCESS_FINE_LOCATION` and `FOREGROUND_SERVICE` permissions are added to your app's permissions automatically. To limit the permissions your bare workflow app requires, see [Excluding Android Permissions](#excluding-android-permissions).
```

File: docs/pages/guides/permissions.md [hunk 7]
```diff
@@ -26,30 +28,40 @@ You can find the full list of available properties in [Apple's InfoPlistKeyRefer
...
 
 Apps using dangerous or signature permissions without valid reasons _may be rejected by Google_. Make sure you follow the [Android permissions best practices](https://developer.android.com/training/permissions/usage-notes) when submitting your app.
 
-#### Excluding Android permissions
+### Excluding Android permissions
+
+When adding Expo and other React Native modules to your project, certain Android permissions might be added automatically. The modules should only add relevant permissions **required** to use the module. However, sometimes you may want to remove some of these permissions.
```
Comment (brentvatne) [inline docs/pages/guides/permissions.md:47]:
```suggestion
Adding packages to your project can result in new Android permissions being added automatically. Ideally, a module should only add the permissions **required** to use the module, and the rest should be opt-in. However, sometimes this isn't the case, and you may want to remove some of these permissions.
```

File: docs/pages/guides/permissions.md [hunk 8]
```diff
@@ -26,30 +28,40 @@ You can find the full list of available properties in [Apple's InfoPlistKeyRefer
...
+
+When you are building your app with [EAS Build](/build/introduction/), you can exclude Android permissions by adding `android.blockedPermissions` in **app.json**. For more information, see the example in the [Android Permissions](/distribution/app-stores/#android-permissions) section in [Deploying to the App Stores](/distribution/app-stores).
```
Comment (brentvatne) [inline docs/pages/guides/permissions.md:49]:
```suggestion
You can exclude Android permissions by adding `android.blockedPermissions` in **app.json**. For more information, see the example in the [Android Permissions](/distribution/app-stores/#android-permissions) section in [Deploying to the App Stores](/distribution/app-stores).
```

Comment (brentvatne) [reply docs/pages/guides/permissions.md:49]: no need to mention building with eas build. this also limits the scope of the information unnecessarily -- because this same mechanism works with `expo prebuild` locally too

File: docs/pages/guides/permissions.md [hunk 9]
```diff
@@ -26,30 +28,40 @@ You can find the full list of available properties in [Apple's InfoPlistKeyRefer
...
-When adding Expo and other React Native modules to your project, certain Android permissions might be implied automatically. The modules should only add relevant permissions **required** to use the module, however, sometimes you may want to remove some of these permissions.
+To limit the permissions your managed workflow app requires, set the `android.permissions` property in your **app.json** to list only the permissions you need, and Expo will also include the minimum permissions it requires to run.
+
+Learn how to configure permissions with the [app manifest `permissions` property](/versions/latest/config/app/#permissions).
+
+</ConfigClassic>
+
+<ConfigReactNative>
 
 Since the `android.permissions` manifest property doesn't work in the bare workflow, you'll need to edit **AndroidManifest.xml** to exclude specific permissions from the build. You can do that with the `tools:node="remove"` attribute on the `<use-permission>` tag.
```
Comment (brentvatne) [inline docs/pages/guides/permissions.md:61]:
```suggestion
Since the `android.blockedPermissions` isn't applied in bare workflow (because **app.json** is not used to configure bare workflow projects), you'll need to edit **AndroidManifest.xml** to exclude specific permissions from the build. You can do that with the `tools:node="remove"` attribute on the `<use-permission>` tag.
```

File: docs/pages/versions/unversioned/sdk/location.md [hunk 10]
```diff
@@ -21,20 +22,31 @@ import SnackInline from '~/components/plugins/SnackInline';
 
 ### Android permissions
 
-- This module requires the permissions for approximate and exact device location. It also needs the foreground service permission to subscribe to location updates, while the app is in use. The `android.permission.ACCESS_COARSE_LOCATION`, `ACCESS_FINE_LOCATION`, and `FOREGROUND_SERVICE` permissions are automatically added.
-- In order to use background location features, you also must add the `android.permission.ACCESS_BACKGROUND_LOCATION` and [submit your app for review and request access to use the background location permission](https://support.google.com/googleplay/android-developer/answer/9799150?hl=en).
+The `expo-location` module requires permission for approximate (`ACCESS_COARSE_LOCATION`) and precise (`ACCESS_FINE_LOCATION`) device location in the `android.permission` property in **app.json**. It also needs foreground service permission (`FOREGROUND_SERVICE`) to subscribe to location updates while the app is in use. When you install this module in your Expo project, all of the three permissions are added automatically.
```
Comment (brentvatne) [inline docs/pages/versions/unversioned/sdk/location.md:25]: this point is confusing to me - it reads like users will need to add these permissions to their app.json, but they are actually added automatically

**Review by brentvatne — DISMISSED** [review]

File: docs/pages/guides/permissions.md [hunk 11]
```diff
@@ -26,30 +28,40 @@ You can find the full list of available properties in [Apple's InfoPlistKeyRefer
...
 
-To limit the permissions your managed workflow app requires, set the `android.permissions` property in your [**app.json** file](../../workflow/configuration.md#android) to list only the permissions you need, and Expo will also include the minimum permissions it requires to run. See the [`Permission types`](#permission-types) below to learn about which Android permissions are added. You can find a full list of all available permissions in the [Android Manifest.permissions reference](https://developer.android.com/reference/android/Manifest.permission).
+Some Expo and React Native modules include permissions by default. For example, if you use `expo-location`, the `ACCESS_COARSE_LOCATION`, `ACCESS_FINE_LOCATION` and `FOREGROUND_SERVICE` permissions will be added to your app automatically. [See the `android.permissions` documentation](/versions/latest/config/app.md#permissions) to learn about which permissions are always included. To limit the permissions your managed workflow app requires, see [Excluding Android Permissions](#excluding-android-permissions).
```
[the remark this answers is not in the document]
Comment (amandeepmittal) [reply docs/pages/guides/permissions.md:33]: I've read the internal linking docs, but it didn't make clear to me about using relative file paths. I thought the arrows were use to provide a use case that this will happen when one uses links.

I guess, I should improve that to be more layman wording for folks like me (taking a note about it).

Thanks! I'll revert the link.

File: docs/pages/versions/unversioned/sdk/location.md (hunk 10 above)
Comment (amandeepmittal) [reply docs/pages/versions/unversioned/sdk/location.md:25]: Good point! If it is confusing, I'll rephrase it to convey what permissions this module automatically adds when using in an Expo project.

**Comment by amandeepmittal** [comment]
Code review changes applied to both the `Permissions` guide and `Location` API reference page.

**Review by EvanBacon — APPROVED** [review]
